workflow/rules/map.smk

```diff
@@ -518,9 +517,11 @@ rule convert_genome_to_sam_oligomap:
         os.path.join(workflow.basedir, "envs", "python.yaml")
     shell:
         "(python {input.script} \
-        -i {input.sort} \
+        {input.sort} \
         -n {params.nh} \
-        > {output.gmap}) &> {log}"
+        --ref {params.ref} \
+        --outdir {params.out_dir} \
+        ) &> {log}"
```

Comment:
Why did you change the API of the script? Why is `--ref` needed now but wasn't before? Not saying it's a bad thing, but you should probably include the reasoning for this in the PR description.

More problematic: Why do we have an output directory now, instead of just a single file? Are you creating more than one output file now? If so, why?

Reply:
If being totally honest, I remember that I made this change for a reason but cannot recall it right now, apologies...

Whatever reason was behind using the output directory, I had to use the `--ref` flag because in order to write to the output file, I needed to give it a name, and depending on the source of the mappings - either genome or transcriptome - the file name changes. Should I simply write the output to the standard output and redirect it from the workflow as before the refactoring?

Reply:
Yes, redirecting the output to `STDOUT` sounds reasonable to me. It's usually the best choice for scripts, because you can use the script in pipes. And it's less code. And overall a more user friendly API, because it requires less specific knowledge on its usage by the user - it will just behave like most other scripts.

Also, you can then name it however you want in the workflow, so no need for the `--ref` as well, I suppose. Besides, if you really want to include a param to take a string to be included in an output file name, consider naming it `--prefix` or something like that. It is much more obvious. Also, set a default.

But anyway, this really only makes sense if you have multiple output files, which should be avoided, if at all possible, for the reasons mentioned above.